test/e2e/cmd/run/job.go

```diff
@@ -73,8 +77,7 @@ func (j *Job) WithResultFile(f string) *Job {
 // onPodEvent ensures that log streaming is started and also manages the internal state of the Job based on the events
 // received from the informer.
 func (j *Job) onPodEvent(client *kubernetes.Clientset, pod *corev1.Pod) {
-	switch pod.Status.Phase { //nolint:exhaustive
-	case corev1.PodRunning:
+	if pod.Status.Phase == corev1.PodRunning {
```

Comment:
Probably just want to `if pod.Status.Phase == corev1.PodRunning && !j.jobStarted {` at this point.

Reply:
Right!